Mathlib/Analysis/InnerProductSpace/Dual.lean

```diff
@@ -69,6 +70,61 @@ variable {E}
 theorem toDualMap_apply {x y : E} : toDualMap 𝕜 E x y = ⟪x, y⟫ :=
   rfl
 
+section NullSubmodule
+
+open SeparationQuotient LinearMap
+
+variable (E)
+
+/-- The null space with respect to the norm. -/
+def nullSubmodule : Submodule 𝕜 E where
```

Comment:
Because you've defined this here, you're assuming that `E` is an inner product space and `𝕜` is `RCLike`, but there's no reason the definition needs to be anywhere near this restrictive. You're going to want to define `nullSubmodule` in a separate file, and then prove the results specific to inner product spaces here. In fact, I think it's only the `ker` results that stay in this file.